document-generator-core/src/main/java/uk/gov/companieshouse/document/generator/core/service/impl/DocumentGeneratorServiceImpl.java

```diff
@@ -0,0 +1,119 @@
+package uk.gov.companieshouse.document.generator.core.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import uk.gov.companieshouse.document.generator.core.document.render.RenderDocumentRequestHandler;
+import uk.gov.companieshouse.document.generator.core.document.render.models.RenderDocumentRequest;
+import uk.gov.companieshouse.document.generator.core.document.render.models.RenderDocumentResponse;
+import uk.gov.companieshouse.document.generator.core.service.DocumentGeneratorService;
+import uk.gov.companieshouse.document.generator.core.service.models.DocumentRequest;
+import uk.gov.companieshouse.document.generator.core.service.models.DocumentResponse;
+import uk.gov.companieshouse.document.generator.interfaces.DocumentInfoService;
+import uk.gov.companieshouse.document.generator.interfaces.model.DocumentInfo;
+import uk.gov.companieshouse.environment.EnvironmentReader;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import static uk.gov.companieshouse.document.generator.core.DocumentGeneratorApplication.APPLICATION_NAME_SPACE;
+
+public class DocumentGeneratorServiceImpl implements DocumentGeneratorService {
+
+    private DocumentInfoService documentInfoService;
+
+    private EnvironmentReader environmentReader;
+
+    private RenderDocumentRequestHandler requestHandler;
+
+    private static final String DOCUMENT_RENDER_SERVICE_HOST_ENV_VAR = "DOCUMENT_RENDER_SERVICE_HOST";
+
+    private static final String CONTEXT_PATH = "/document-render/store?is_public=true";
+
+    private static final Logger LOG = LoggerFactory.getLogger(APPLICATION_NAME_SPACE);
+
+    @Autowired
+    public DocumentGeneratorServiceImpl(DocumentInfoService documentInfoService, EnvironmentReader environmentReader,
+                                        RenderDocumentRequestHandler requestHandler) {
+
+        this.documentInfoService = documentInfoService;
+        this.environmentReader = environmentReader;
+        this.requestHandler = requestHandler;
+    }
+
+    @Override
+    public DocumentResponse generate(DocumentRequest documentRequest) {
+
+        DocumentInfo documentInfo = null;
```

Comment:
Do these need to be instantiated as null explicitly?

Reply:
Yes